app/models/katello/content_view_history.rb

```diff
@@ -66,5 +66,13 @@ def humanized_status
         _("Success")
       end
     end
+
+    def description
+      if self.task.try(:label) == "Actions::Katello::ContentView::Publish"
+        content_view_version.description
```

Comment:
I would lean towards #1 as well; however, I wonder under what conditions, if any, it could  occur.  AFAIK, katello doesn't provide provide users with a direct way to delete the history records.  It looks like perhaps they can be deleted by either deleting a lifecycle environment or a content view version, in which case my assumption is that the version will no longer exist either.